packages/core/test/scuttle.spec.js

```diff
@@ -0,0 +1,126 @@
+const test = require('ava')
+const { evaluateWithSourceUrl } = require('./util')
+const { scuttle } = require('../src/scuttle')
+
+const SCUTTLER_NAME_ERROR = {
+  message:
+    'LavaMoat - \'scuttlerName\' function "SCUTTLER" expected on globalRef.' +
+    'To learn more visit https://github.com/LavaMoat/LavaMoat/pull/462.',
+}
+
+const err = (intrinsic) =>
+  'LavaMoat - property "' +
+  intrinsic +
+  '" of globalThis is inaccessible under ' +
+  'scuttling mode. To learn more visit https://github.com/LavaMoat/LavaMoat/pull/360.'
+
+function testScuttling(
+  globalRef,
+  intrinsics = ['Array', 'Int8Array', 'RegExp']
+) {
+  return intrinsics.some((intrinsic) => {
+    try {
+      const _ = globalRef[intrinsic]
+      return false
+    } catch (e) {
+      if (err(intrinsic) === e.message) {
+        return true
+      }
+      throw new Error('Unknown error thrown at testScuttling: ' + e.message)
+    }
+  })
+}
+
+test('scuttle - no opts', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis)
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+})
+
+test('scuttle - opts as bool', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, false)
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, true)
+  t.is(
+    testScuttling(vmGlobalThis),
+    true,
+    'expect intrinsics to not be accessible'
+  )
+})
+
+test('scuttle - opts as object', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, { enabled: false })
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, { enabled: true })
+  t.is(
+    testScuttling(vmGlobalThis),
+    true,
+    'expect intrinsics to not be accessible'
+  )
+})
+
+test('scuttle - exceptions', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, {
+    enabled: true,
+    exceptions: ['/[a-zA-Z0-8]*Array/', 'RegExp'],
```

Comment:
seems like we could avoid requiring the forward-slashes?

why do we need to support strings anyhow?

Reply:
These have to be primitive values, because in the browser version, this gets bundled

Reply:
I don't understand why bundling would mean we cannot use a `RegExp`.

Why is the character class `[a-zA-Z0-8]` and not `[a-zA-Z0-9]`?

(I'm asking all these questions because I don't know anything about any of this)

Reply:
`0-8` looks like a typo

Reply:
You just bombed attackers search for LavaMoat-scuttling protected programs with `window['999']` exposed API sir (fixed 8a71f8464)

Reply:
For the browser version of LavaMoat, where the program gets bundled along with our protection (e.g. lavamoat-browserify), this means configurable arguments that need to be introduced into the output must be primitive values, otherwise how are you going to ship them into the bundled output?

For example, if it was:

```javascript
{
    enabled: true,
    exceptions: [new RegExp('/[a-zA-Z0-8]*Array/'), new RegExp('RegExp')],
}
```

You'd end up with an output similar to this:

```javascript
// lavamoat
const args = {"enabled":true,"exceptions":[{},{}]};
function scuttle() {
...

// program
alert('this program is protected by lavamoat');
```

if that makes sense?